### PR TITLE
Make compost usable by the farmer

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Allow farmer to use and request compost/bonemeal
- Move some pick up code to abstractEntityAiBasic to be used by all.

Review please

### CODE REVIEW

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 40656 |  0  |  0  |  0   |
| Errors |  675  |  0  |  0  |  0   |
##### More Information:

File: src/api/java/com/minecolonies/api/util/InventoryUtils.java [hunk 1]
```diff
@@ -179,6 +180,25 @@ public static int findFirstSlotInItemHandlerWith(@NotNull final IItemHandler ite
...
+     * Shrinks a specific stack in an item handler.
+     * @param itemHandler the handler..
+     * @param itemStackSelectionPredicate the predicate..
+     * @return true if successful.
+     */
+    public static boolean shrinkItemCountInItemHandler(final IItemHandler itemHandler, @NotNull final Predicate<ItemStack> itemStackSelectionPredicate)
+    {
+        for (int slot = 0; slot < itemHandler.getSlots(); slot++)
+        {
+            if (itemStackSelectionPredicate.test(itemHandler.getStackInSlot(slot)))
```
Comment (marchermans) [inline src/api/java/com/minecolonies/api/util/InventoryUtils.java]: Insert an none empty check here. It makes no sense to call the predicate with an Empty stack when we are shrinking.

**Review by marchermans — CHANGES_REQUESTED** [review]
A small change to the Inventory API would be nice.

**Review by marchermans — DISMISSED** [review]
Nice way to solve the not null check :D

**Pull request merged**